src/main/java/bio/terra/service/snapshot/SnapshotTableDao.java

```diff
@@ -93,42 +111,59 @@ private void createColumns(UUID tableId, Collection<Column> columns) {
   }
 
   public List<SnapshotTable> retrieveTables(UUID parentId) {
-    MapSqlParameterSource params = new MapSqlParameterSource().addValue("parent_id", parentId);
-    return jdbcTemplate.query(
+    MapSqlParameterSource params = new MapSqlParameterSource().addValue("snapshot_id", parentId);
+    Map<UUID, SnapshotTable> tableMap = new TreeMap<>();
+    jdbcTemplate.query(
         sqlSelectTable,
         params,
         (rs, rowNum) -> {
+          UUID tableId = rs.getObject("table_id", UUID.class);
           SnapshotTable table =
-              new SnapshotTable()
-                  .id(rs.getObject("id", UUID.class))
-                  .name(rs.getString("name"))
-                  .rowCount(rs.getLong("row_count"));
-          List<Column> columns = retrieveColumns(table);
-          table.columns(columns);
+              tableMap.computeIfAbsent(
+                  tableId,
+                  (id) -> {
+                    try {
+                      return new SnapshotTable()
+                          .id(id)
+                          .name(rs.getString("table_name"))
+                          .rowCount(rs.getLong("table_row_count"))
+                          .columns(new ArrayList<>())
+                          .primaryKey(new ArrayList<>());
+                    } catch (SQLException e) {
+                      throw new RuntimeException(e);
+                    }
+                  });
 
-          Map<String, Column> columnMap =
-              columns.stream().collect(Collectors.toMap(Column::getName, Function.identity()));
+          // Add column to table
+          Column column = getColumnRowMapper(table).mapRow(rs, rowNum);
+          table.getColumns().add(column);
 
-          List<String> primaryKey = DaoUtils.getStringList(rs, "primary_key");
-          List<Column> naturalKeyColumns =
-              primaryKey.stream().map(columnMap::get).collect(Collectors.toList());
-          table.primaryKey(naturalKeyColumns);
+          // Add primary key to table if appropriate
+          List<String> primaryKey = DaoUtils.getStringList(rs, "table_primary_key");
```

Comment:
This is the same for all rows in a table but is recomputed for each row. If you wanted to improve this, you could use another `Map.computeIfAbsent()` for `primaryKey`.

Reply:
Yeah, I didn't know if it was worth the code complexity.  I can add it

Reply:
Actually, this is good.  I can use this to preserve the order of the primary key columns (probably not super important but might as well)